File: app/src/main/java/tw/y_studio/ptt/API/BaseAPIHelper.java
package tw.y_studio.ptt.API;

import android.content.Context;

import okhttp3.OkHttpClient;
import tw.y_studio.ptt.BuildConfig;
import tw.y_studio.ptt.Utils.OkHttpUtils;

public class BaseAPIHelper {
    protected String hostUrl = BuildConfig.API_Domain;
    protected OkHttpClient mOkHttpClient;

    public BaseAPIHelper(Context context) {
        mOkHttpClient = new OkHttpUtils().getTrustlAllClient(context);
    }

    public void close() {
        if (mOkHttpClient != null) {
            mOkHttpClient.dispatcher().cancelAll();
        }
        mOkHttpClient = null;
    }
}

File: app/src/main/java/tw/y_studio/ptt/api/BaseAPIHelper.kt
package tw.y_studio.ptt.api

import android.content.Context
import okhttp3.OkHttpClient
import tw.y_studio.ptt.BuildConfig
import tw.y_studio.ptt.utils.OkHttpUtils

abstract class BaseAPIHelper : IBaseAPI {
    constructor() {
        okHttpClient = OkHttpUtils().trustAllClient
    }
    constructor(context: Context?) {
        okHttpClient = OkHttpUtils().trustAllClient
    }

    @JvmField
    protected val hostUrl = BuildConfig.APIDomain

    protected val okHttpClient: OkHttpClient?
    override fun close() {
        okHttpClient?.dispatcher?.cancelAll()
    }
}
